torchdynamo/variables/misc.py

```diff
@@ -262,14 +262,19 @@ def call_function(
 
 
 class GradModeVariable(ContextWrappingVariable):
+    @staticmethod
+    def create(tx, target_value, initial_value=None, **kwargs):
+        var = GradModeVariable(target_value, initial_value, **kwargs)
```

Comment:
initial value is really weird, because its None here, and then used to early return out of _call_func, but `_ initial_value()` just gets the current state of `torch.is_grad_enabled()` Which gets called in the super through:

```
if initial_value is None:
            initial_value = self._initial_value()
        self.initial_value = initial_value
```

And we never actually set it.

Can we just:

1) remove it from the ctor of this class
2) Fix the `ContextWrappingVariable` ctor to not call self._initial_value()
3) Instead have `ContextWrappingVariable` just take a value

It also look like its why we have a ContextManagerVariable and a ContextWrappingVariable - two confusing names, one of which is the base (ContextManagerVariable) which acts as a past-through for the of `FakeContextWrappingVariable` (as it cannot call super, due to not having an initial value).

Other nits:

- FakeContextWrappingVariable is too tightly coupled to `"autograd.profiler.profile"` for its name 

- ContextWrappingVariable being used for something that is not a context (set grad enabled in this case)

Reply:
Ah, its a mess. We have incrementally added fixes for grad support, and code is messy now. We need a refactor.

Regarding the initial value, whats the key concern? I want to point out that we do need `initial_value` for the `exit` of the context manager. Initial value just keeps track of the initial torch grad state, and uses it during the `exit` call.

We do not really set the `initial_state` during the `enter` or `init` function, because we don't have to. For the export usecase, I would assume that there would be a guard that checks that the torch initial state is not different from the exported graph state.